airflow_providers_wherobots/hooks/rest_api.py

```diff
@@ -0,0 +1,81 @@
+"""
+Hook for WhereRobots API
+"""
+
+from functools import cached_property
+from typing import Any, Optional
+
+import requests
+from airflow.hooks.base import BaseHook
+from airflow.models import Connection
+from requests import PreparedRequest
+from requests.adapters import HTTPAdapter, Retry
+from requests.auth import AuthBase
+
+from airflow_providers_wherobots.hooks.base import DEFAULT_CONN_ID
+from airflow_providers_wherobots.wherobots.models import Run, CreateRunPayload
+
+
+class WherobotsAuth(AuthBase):
+    def __init__(self, api_key: str):
+        self.api_key = api_key
+
+    def __call__(self, r: PreparedRequest):
+        # modify and return the request
+        r.headers["Authorization"] = "Bearer " + self.api_key
+        return r
+
+
+class WherobotsRestAPIHook(BaseHook):
+    conn_name_attr = "wherobots_conn_id"
+    default_conn_name = DEFAULT_CONN_ID
+
+    def __init__(
+        self,
+        wherobots_conn_id: str = DEFAULT_CONN_ID,
+        retry_limit: int = 3,
+        retry_min_delay: float = 1.0,
+    ):
+        super().__init__()
+        self.wherobots_conn_id = wherobots_conn_id
+        self.retry_limit = retry_limit
+        self.retry_min_delay = retry_min_delay
+
+    @cached_property
+    def conn(self) -> Connection:
+        return self.get_connection(self.wherobots_conn_id)
+
+    def _api_call(
+        self,
+        method: str,
+        endpoint: str,
+        payload: Optional[dict[str, Any]] = None,
+    ) -> Any:
+        retries = Retry(
+            total=self.retry_limit,
+            backoff_factor=self.retry_min_delay,
+            status_forcelist=[500, 502, 503, 504],
+        )
```

Comment:
Seems like you could create this (and the HTTPAdapter) only once in the constructor